### PR TITLE
cli: `fmt -check` should return early on diff

### PR DESCRIPTION
**Pull request by tgross**
Fixes https://github.com/hashicorp/nomad/issues/16170

The `nomad fmt -check` command incorrectly writes to file because we didn't
return before writing the file on a diff. Fix this bug and update the command
internals to differentiate between the write-to-file and write-to-stdout code
paths, which are activated by different combinations of options and flags.

The docstring for the `-list` and `-write` flags is also unclear and can be
easily misread to be the opposite of the actual behavior. Clarify this and fix
up the docs to match.

This changeset also refactors the tests quite a bit so as to make the test
outputs clear when something is incorrect and avoid reusing the UI state.

### CODE REVIEW

File: command/fmt_test.go [hunk 1]
```diff
@@ -18,80 +19,115 @@ func TestFmtCommand(t *testing.T) {
...
+			expectWrite: true,
+			expectCode:  0,
+		},
+	}
 
-			input, err := os.ReadFile(inFile)
-			require.NoError(t, err)
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			inFile := filepath.Join("testdata", "fmt", tc.testFile+inSuffix)
```
Comment (angrycub) [inline command/fmt_test.go:62]: Maybe we could parallelize these cases? If so, we can scratch the tmpDir on line 23
```suggestion
			tc := tc
			ci.Parallel(t)
			tmpDir := t.TempDir()
			inFile := filepath.Join("testdata", "fmt", tc.testFile+inSuffix)
```

File: command/fmt_test.go [hunk 2]
```diff
@@ -18,80 +19,115 @@ func TestFmtCommand(t *testing.T) {
...
 	const inSuffix = ".in.hcl"
 	const expectedSuffix = ".out.hcl"
-	tests := [][]string{
-		{"nomad", "-check"},
-		{"nomad", ""},
-		{"job", "-check"},
-		{"job", ""},
-	}
+
 	tmpDir := t.TempDir()
```
Comment (angrycub) [inline command/fmt_test.go:23]: (see later suggestion) Should create the tempdir in the test case if we're parallelizing them
```suggestion
```

File: website/content/docs/commands/fmt.mdx [hunk 3]
```diff
@@ -46,9 +49,9 @@ client {
 }
 
 $ nomad fmt
-
 agent.hcl
-$ cat agent.hcl 
+
+$ cat agent.hcl
```
Comment (angrycub) [inline website/content/docs/commands/fmt.mdx:54]: Should be its own shell-session block

File: website/content/docs/commands/fmt.mdx [hunk 4]
```diff
@@ -46,9 +49,9 @@ client {
 }
 
 $ nomad fmt
```
Comment (angrycub) [inline website/content/docs/commands/fmt.mdx:51]: Should be its own shell session block

**Review by angrycub — APPROVED** [review]
Needs a couple of small touches to the docs markup, and I shared some testing thoughts that are totally optional.

**Review by shoenig — APPROVED** [review]
LGTM

**Pull request merged**